src/components/ModalWithForm.js

```diff
@@ -18,6 +19,14 @@ export default class ModalWithForm extends Modal {
     return formValues;
   }
 
+  setLoading(isLoading, text) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = text;
+    }
+  }
+
   setEventListeners() {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();

```

Comment:
Please, delete ` this.close();` from line 34


* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well